core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java

```diff
@@ -102,9 +100,9 @@ public Assignments(@JsonProperty("assignments") Map<Symbol, Expression> assignme
         this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
     }
 
-    public List<Symbol> getOutputs()
+    public Set<Symbol> getOutputs()
```

Comment:
The order implied by the list is significant.

Reply:
@martint can you explain to me where it matters?
(outside of LocalExecPlanner, which probably could go away without having it)

Reply:
It's used to establish the order of the outputs of ProjectNode and others, which should be stable and have an explicit order (this is not *quite* the case today, but as part of the AST/IR refactoring, the plan is to make the output of a plan node follow its type -- i.e., a row type, where field order is significant)

Reply:
Yet the underlying collection is a set, so how does that change make any difference?  Since Guava immutable sets/maps do indeed preserve order, we can just add javadoc comment about it, if that is really necessary

Reply:
It's a set with guaranteed traversal order. Returning a regular `Set` does not convey that and we don't want to return `ImmutableSet`. A javadoc comment is easy to overlook.

Let me take a look at who depends on this to see if it's even reasonable to not have guaranteed traversal order.